app/src/main/java/ch/epfl/sdp/firebase/db/FirestoreDatabase.java

```diff
@@ -0,0 +1,29 @@
+package ch.epfl.sdp.firebase.db;
+
+import androidx.annotation.NonNull;
+
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import ch.epfl.sdp.db.Database;
+import ch.epfl.sdp.firebase.db.queries.FirebaseCollectionQuery;
+import ch.epfl.sdp.db.queries.CollectionQuery;
+
+public class FirestoreDatabase implements Database {
+
+    private final FirebaseFirestore mDb;
+
+    public FirestoreDatabase(@NonNull FirebaseFirestore firestore) {
+        if(firestore == null) {
+            throw new IllegalArgumentException();
+        }
+        mDb = firestore;
+    }
+
+    @Override
+    public CollectionQuery query(@NonNull String collection) {
+        if(collection == null) {
+            throw new IllegalArgumentException();
+        }
+        return new FirebaseCollectionQuery(mDb, mDb.collection(collection));
+    }
```

Comment:
Pour un prochain sprint, peut-on eviter ces boilerplate de toujours test for les null values. Peut être ajouter une dependency avec des @NonNull qui throw si Null et n'est pas juste un hint pour l'ide comme c'est le cas pour l'instant.

Reply:
Idée intéressante mais on devra quand même faire le test en question pour satisfaire le coverage.